clusters/data_node_type.go

```diff
@@ -89,6 +94,19 @@ type NodeType struct {
 	Graviton              bool                          `json:"is_graviton,omitempty"`
 }
 
+func (nt NodeType) shouldBeSkipped() bool {
+	shouldBeSkipped := false
+	if nt.NodeInfo != nil {
+		for _, st := range nt.NodeInfo.Status {
+			if st == CloudProviderNodeStatusNotAvailableInRegion || st == CloudProviderNodeStatusNotEnabled {
+				shouldBeSkipped = true
+				break
+			}
+		}
+	}
```

Comment:
should we also check `node_info.available_core_quota >= num_cores`

Reply:
not sure if it will help a lot, as we're working on the individual node level...

Reply:
I suppose it depends on the use-case. For mounts, it will help since we only start one node always and use it for all mounts. When the use-case is to for user-defined clusters, it's less clear how much it would help since the number of cluster and number of executors is unknown at this stage.

Reply:
comment doesn't block merge